### PR TITLE
Added numpy.random.laplace and test

### LINKED ISSUE #19717: laplace

### PR DESCRIPTION
**Pull request by RITIK-12**
close #19717

### CODE REVIEW

**Review by AryanSharma21 — COMMENTED** [review]
Hello @RITIK-12 please mention `close #issue_number` in the PR description so that it will automatically close the related issue from the todo list . 🙂

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by RITIK-12** [comment]
> Hello @RITIK-12 please mention `close #issue_number` in the PR description so that it will automatically close the related issue from the todo list . 🙂

Done, thanks :)

**Comment by Sam-Armstrong** [comment]
ivy-gardener

File: ivy/functional/frontends/numpy/random/functions.py [hunk 1]
```diff
@@ -197,3 +197,17 @@ def logistic(loc=0.0, scale=1.0, size=None):
...
     x = loc + scale * ivy.log(u / (1 - u))
     return x
+
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def laplace(loc=0.0, scale=1.0, size=None):
+    u = ivy.random_uniform(low=0.0, high=0.0, shape=size, dtype="float64")
+
+    if u < 0.5:
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/numpy/random/functions.py:207]: using an if statement like this doesn't make sense on arrays with multiple values. just looking at the probability density function in the docs (https://numpy.org/doc/stable/reference/random/generated/numpy.random.laplace.html) I don't think you need to include any conditionals like this? please can you try to fix this and make sure your test passes (which you can run with pytest ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py::test_numpy_laplace) 🙂

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]

**Comment by Sam-Armstrong** [comment]
also, remember to pull this branch as the code has been linted 😁

File: ivy/functional/frontends/numpy/random/functions.py (hunk 1 above)
Comment (RITIK-12) [reply ivy/functional/frontends/numpy/random/functions.py]: Hey @Sam-Armstrong, 

Thanks for your suggestion :)

I have updated the function. The tests were successful.

File: ivy/functional/frontends/numpy/random/functions.py [hunk 2]
```diff
@@ -197,3 +197,11 @@ def logistic(loc=0.0, scale=1.0, size=None):
     u = ivy.random_uniform(low=0.0, high=0.0, shape=size, dtype="float64")
     x = loc + scale * ivy.log(u / (1 - u))
     return x
+
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def laplace(loc=0.0, scale=1.0, size=None):
+    u = ivy.random_uniform(low=0.0, high=1.0, shape=size, dtype="float64")
+    x = ivy.exp(-abs(u - loc) / scale)/ (2 * scale)
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/numpy/random/functions.py:206]: This implementation is incorrect, I believe the correct implementation would look something like this:
```
    u = ivy.random_uniform(low=-0.5, high=0.5, shape=size)
    x = loc - (scale * ivy.sign(u) * ivy.log(1 - (2 * ivy.abs(u))))
```
Try sampling the frontend function to see the difference, and compare the results to the native numpy function.

File: ivy/functional/frontends/numpy/random/functions.py [hunk 3]
```diff
@@ -197,3 +197,11 @@ def logistic(loc=0.0, scale=1.0, size=None):
...
+    return x
```
Comment (Sam-Armstrong) [inline ivy/functional/frontends/numpy/random/functions.py:207]: We need a line of whitespace at the end of the file

Comment (RITIK-12) [reply ivy/functional/frontends/numpy/random/functions.py]: > 

@Sam-Armstrong Thanks a lot. Due to branch conflicts, the older commits were deleted. I have generated this [PR](https://github.com/unifyai/ivy/pull/21747) with the changes you suggested. Please review It.

File: ivy/functional/frontends/numpy/random/functions.py (hunk 2 above)
Comment (RITIK-12) [reply ivy/functional/frontends/numpy/random/functions.py]: Hey @Sam-Armstrong ,

Thank you so much for pointing it out. This function is closest to Numpy's laplace function. I tried using the code mentioned below, it also works fine, but yours seems more intuitive. 

`def laplace(loc=-0.0, scale=1.0, size=None):
Screenshot 2023-08-12 at 1 17 14 PM

    u = ivy.random_uniform(low=0.0, high=1.0, shape=size)
    u_normalized = u - 0.5
    x = loc - (scale * ivy.sign(u_normalized) * ivy.log(1 - (2 * ivy.abs(u_normalized))))
    return x
`
I tried s
Screenshot 2023-08-12 at 1 17 46 PM
sampling both the functions and compared it with numpy's. I have attached the screenshot.

I have made the changes, but the commits got deleted due to conflicts. I had to create a [new PR](https://github.com/unifyai/ivy/pull/21747). Please review it and let me know your suggestions.

**Pull request closed without merging**